src/inet/TCPEndPointImplLwIP.cpp

```diff
@@ -130,18 +121,16 @@ CHIP_ERROR TCPEndPointImplLwIP::ConnectImpl(const IPAddress & addr, uint16_t por
             return res;
     }
 
-    // Lock LwIP stack
-    LOCK_TCPIP_CORE();
-
     res = GetPCB(addrType);
 
     if (res == CHIP_NO_ERROR)
     {
-        tcp_arg(mTCP, this);
-        tcp_err(mTCP, LwIPHandleError);
-
         ip_addr_t lwipAddr = addr.ToLwIPAddr();
-        res                = chip::System::MapErrorLwIP(tcp_connect(mTCP, &lwipAddr, port, LwIPHandleConnectComplete));
+        res                = chip::System::MapErrorLwIP(RunOnTCPIPRet([this, &lwipAddr, port]() {
+            tcp_arg(mTCP, this);
+            tcp_err(mTCP, LwIPHandleError);
+            return tcp_connect(mTCP, &lwipAddr, port, LwIPHandleConnectComplete);
+        }));
 
         // Ensure that TCP timers are started
```

Comment:
Does start_tcp_timers below not need RunOnTCPIPRet?

Reply:
No, it calls tcpip_callback which is a thread-safe function.